### PR TITLE
hangups: 0.4.4

### PR DESCRIPTION
**Pull request by aswan89**
###### Motivation for this change

Addition of hangups application/library and required dependencies to nixpkgs.
Allows use of Google hangouts in terminal.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{ pkgs, stdenv, python3 }:
```
Comment (FRidh) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: Don't pass in `pkgs`

File: pkgs/development/python-modules/ReParser/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+buildPythonPackage rec {
+  pname = "ReParser";
+  version = "1.4.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0nniqb69xr0fv7ydlmrr877wyyjb61nlayka7xr08vlxl9caz776";
+  };
+
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/ReParser/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/readlike/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "readlike";
+  version = "0.1.2";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1ck65ycw51f4xnbh4sg9axgbl81q3azpzvd5f2nvrv2fla9m8r08";
+  };
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/readlike/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+let 
+  packageOverrides = self: super:{
```
Comment (FRidh) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: :+1:

**Review by FRidh — CHANGES_REQUESTED** [review]
Please also check the contribution guidelines regarding the naming of commits.

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      src = oldAttrs.src.override {
+        inherit version;
+        extension = "zip";
+        sha256 = "0ka3nximli7wcpy381grn6d75fnpf91050iwdcj4shf53z0m1fg2";
+      };
+    }); 
+  };
+  pythonPackages = (python3.override { inherit packageOverrides;}).pkgs;
+
+in python3.pkgs.buildPythonApplication rec {
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: `pythonPackages`

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    MechanicalSoup = super.MechanicalSoup.overridePythonAttrs (oldAttrs: rec {
+      version = "0.6.0";
+      src = oldAttrs.src.override {
+        inherit version;
+        extension = "zip";
+        sha256 = "0ka3nximli7wcpy381grn6d75fnpf91050iwdcj4shf53z0m1fg2";
+      };
+    }); 
+  };
+  pythonPackages = (python3.override { inherit packageOverrides;}).pkgs;
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: Please add a space after `;`

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    readlike
+    MechanicalSoup
+  ];
+
+  src = pythonPackages.fetchPypi {
+    inherit pname version;
+    sha256 = "1sm8c9mfcdyhlb6k2cq1r4j3zpc4z47z7zs8yycmf45nmv0vg0h9";
+  };
+ 
+  doCheck = true;
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: not necessary, is set automatically

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.readlike python3.pkgs.readlike python2.pkgs.ReParser python3.pkgs.ReParser hangups

**Comment by dotlambda** [comment]
I think you need to set `disabled = !isPy3k` for `ReParser`.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.readlike python3.pkgs.readlike python3.pkgs.ReParser hangups

**Comment by dotlambda** [comment]
> Ran 0 tests in 0.000s

Please add appropriate `checkPhase`s.

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  checkInputs = [ 
+    pythonPackages.docutils
+    pythonPackages.pytest
+  ];
+  
+  checkPhase = ''
+    python3 setup.py check --metadata --restructuredtext --strict
+  '';
+
+  doCheck = true;
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: not needed

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    readlike
+    MechanicalSoup
+  ];
+
+  src = pythonPackages.fetchPypi {
+    inherit pname version;
+    sha256 = "1sm8c9mfcdyhlb6k2cq1r4j3zpc4z47z7zs8yycmf45nmv0vg0h9";
+  };
+
+  checkInputs = [
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: please use `with pythonPackages`

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.readlike python3.pkgs.readlike python3.pkgs.ReParser hangups

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+  
+  checkPhase = ''
+    python3 setup.py check --metadata --restructuredtext --strict
+  '';
+
+  doCheck = true;
+ 
+  meta = with stdenv.lib; {
+    description = "The first third-party instant messaging client for Google Hangouts";
+    homepage = http://github.com/tdryer/hangups;
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: https

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    MechanicalSoup
+  ];
+
+  src = pythonPackages.fetchPypi {
+    inherit pname version;
+    sha256 = "1sm8c9mfcdyhlb6k2cq1r4j3zpc4z47z7zs8yycmf45nmv0vg0h9";
+  };
+
+  checkInputs = [ 
+    pythonPackages.docutils
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: probably not needed

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    pythonPackages.pytest
+  ];
+  
+  checkPhase = ''
+    python3 setup.py check --metadata --restructuredtext --strict
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: That doesn't really test anything. Something like
```
checkPhase = ''
  py.test
'';
```
should work. However, you'll need to use `fetchFromGitHub` as the tests are not included in the PyPI tarball.

File: pkgs/development/python-modules/readlike/default.nix [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "readlike";
+  version = "0.1.2";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1ck65ycw51f4xnbh4sg9axgbl81q3azpzvd5f2nvrv2fla9m8r08";
+  };
+
```
Comment (dotlambda) [inline pkgs/development/python-modules/readlike/default.nix]: `checkPhase` needed

File: pkgs/development/python-modules/ReParser/default.nix [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+buildPythonPackage rec {
+  pname = "ReParser";
+  version = "1.4.3";
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0nniqb69xr0fv7ydlmrr877wyyjb61nlayka7xr08vlxl9caz776";
+  };
+
```
Comment (dotlambda) [inline pkgs/development/python-modules/ReParser/default.nix]: Please set `doCheck = false` and add a comment that there are no tests.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/instant-messengers/hangups/default.nix [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+  disabled = !pythonPackages.isPy3k;
+
+  propagatedBuildInputs = with pythonPackages; [ 
+    ConfigArgParse 
+    requests
+    aiohttp
+    ReParser
+    appdirs
+    urwid
+    protobuf3_1
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/hangups/default.nix]: Please use `protobuf` and pin the version as for the packages above. You'll have to use
```
protobuf = super.protobuf.override {
  protobuf = pkgs.protobuf3_1;
};
```
however.

**Pull request closed without merging**